pcapgo/read.go

```diff
@@ -101,6 +101,9 @@ func (r *Reader) ReadPacketData() (data []byte, ci gopacket.CaptureInfo, err err
 	}
 
 	var n int
+	if 16+ci.CaptureLength > len(r.buf) {
+		return
```

Comment:
Should return an error.  Note that if this happens, it's a malformed packet file, since the file said the snaplen was X and is now offering up a packet that's X+N.  Maybe have the error say that.